channel/consensus_channel/consensus_channel.go

```diff
@@ -587,19 +565,19 @@ func (p *Proposal) Target() types.Destination {
 type SignedProposal struct {
 	state.Signature
 	Proposal Proposal
+	TurnNum  uint64
```

Comment:
This is the key change, by moving the `TurnNum` to the `SignedProposal` means we only have to deal with a turn number when we are signing/validating a signed proposal.

This lets the follower easily check if a proposal matches what they're expecting without specifying a turn number. This means the leader can solely be responsible for choosing the turn order for proposals.

We can still use the `TurnNum` to sort the proposals so they are returned in order.

Reply:
Main concern: is the turnNum still implicitly part of the data endorsed by the `state.Signature` here, or has this change introduced opportunity for persons to "spoof" turn numbers in some way.

My observation: looks fine / safe.

The signature generated in `Propose()` is on the the updated `vars` object, which has handled the proposal and bumped its turnNumber.

Reply:
> Main concern: is the turnNum still implicitly part of the data endorsed by the `state.Signature` here, or has this change introduced opportunity for persons to "spoof" turn numbers in some way.

The turn number is still part of the `state.Signature` so we're still validating the turn number and preventing spoofing.